eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailCommand.java

```diff
@@ -0,0 +1,265 @@
+            return;
+        }
+
+        this.prisonerService.releasePlayer(target, player);
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailReleaseSender())
+            .placeholder("{PLAYER}", target.getName())
+            .player(player.getUniqueId())
+            .send();
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailReleaseBroadcast())
+            .placeholder("{PLAYER}", target.getName())
+            .all()
+            .send();
+    }
+
+    @Execute(name = "release -all", aliases = { "release *" })
+    @Permission("eternalcore.jail.release")
+    @DescriptionDocs(description = "Release all players from jail")
+    void executeJailReleaseAll(@Context Player player) {
+        if (this.prisonerService.getPrisoners().isEmpty()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailReleaseNoPlayers())
+                .player(player.getUniqueId())
+                .send();
+            return;
+        }
+
+        this.prisonerService.releaseAllPlayers();
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailReleaseAll())
+            .all()
+            .send();
+    }
+
+    @Execute(name = "list")
+    @Permission("eternalcore.jail.list")
+    @DescriptionDocs(description = "List all jailed players")
+    void executeJailList(@Context Player player) {
+        if (this.prisonerService.getPrisoners().isEmpty()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailListEmpty())
+                .player(player.getUniqueId())
+                .send();
+            return;
+        }
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailListHeader())
+            .player(player.getUniqueId())
+            .send();
+
+        for (Prisoner prisoner : this.prisonerService.getPrisoners()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailListPlayerEntry())
+                .placeholder("{PLAYER}", this.server.getOfflinePlayer(prisoner.getPlayerUniqueId()).getName())
+                .placeholder("{DURATION}", DurationParser.TIME_UNITS.format(prisoner.getRemainingTime()))
+                .placeholder("{DETAINED_BY}", prisoner.getDetainedBy())
+                .player(player.getUniqueId())
+                .send();
+        }
+    }
+
+    private boolean isPrisonAvailable(Player player, Player target) {
+        if (!this.jailService.isJailLocationSet()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailLocationNotSet())
+                .player(player.getUniqueId())
+                .send();
+            return true;
+        }
+
+        if (!target.isOnline()) {
```

Comment:
Remove this check. `Player` can't be offline in the same server tick. If you get actual Player instance, it is not possible to receive isOffline -> true